client/src/pages/sports-grid/grid.js

```diff
@@ -1,45 +1,57 @@
 'use client'
 
 import Table from 'react-bootstrap/Table';
-import TeamSelector from './team-selector';
-import PlayerInput from './player-input';
+import Spinner from 'react-bootstrap/Spinner';
+
 import React from 'react';
 
+import PlayerInput from './player-input';
+import TeamLogo from './team-logo';
+
+import { useFetch } from '../../utils/hooks';
+
 export default function Grid() {
+    const data = useFetch(
+        'http://localhost:8000/sports-grid/basketball'
+    );
 
-    return (
-        <Table variant="dark">
-            <thead>
-                <tr>
-                    <td></td>
-                    <td>
-                        <TeamSelector></TeamSelector>
-                    </td>
-                    <td>
-                        <TeamSelector></TeamSelector>
-                    </td>
-                    <td>
-                        <TeamSelector></TeamSelector>
-                    </td>
-                </tr>
-                <tr>
-                    <td><TeamSelector></TeamSelector></td>
-                    <td><PlayerInput></PlayerInput></td>
-                    <td><PlayerInput></PlayerInput></td>
-                    <td><PlayerInput></PlayerInput></td>
-                </tr>
+    const tableHeaders = [];
+    const tableRows = [];
+
+    if (!data) {
+        return(<div>
+            <Spinner animation="border" role="status">
+                <span className="visually-hidden">Loading...</span>
+            </Spinner>
+        </div>)
+    }
+
+    if (!!data.cols) {
```

Comment:
i personally don't like this. i much prefer explicit checks, especially if server returns something unexpected this may still pass. i think we should validate the input separately, return early if we dont have the right input (maybe have a error state as a separate issue).